### ISSUE TITLE
removing one fqdn subordinate removes all nginx sites

### ISSUE THREAD

**Issue by kwmonroe** [tengu-team/layer-ssl-termination-proxy#19]
I had multiple `ssl-termination-fqdn-5` charms deployed and related similar to this:

```bash
juju deploy cs:~tengu-team/ssl-termination-proxy-15 proxy
juju deploy cs:~tengu-team/ssl-termination-fqdn-5 fqdn1
juju deploy cs:~tengu-team/ssl-termination-fqdn-5 fqdn2
juju deploy cs:~tengu-team/ssl-termination-fqdn-5 fqdn3

juju deploy ghost g1
juju deploy ghost g2
juju deploy ghost g3

juju config [fqdnX] [fqdns=X.domain.com]
juju relate [fqdnX] [gX]
juju relate proxy [fqdnX]
```

I ran `juju remove-relation fqdn1 g1; juju remove-application fqdn1` and soon realized that the proxy had stopped serving requests for `g2` and `g3` as well.  In fact, all the `/etc/nginx/sites-available/ssl-termination/*` configuration for those sites had been removed.

I think this is what happened:
1. `fqdn1` was no longer related to a website, and therefore [set empty cert info](https://github.com/tengu-team/layer-ssl-termination-fqdn/blob/737faf0041a0628630f00121119cb15334c95933/reactive/ssl-termination-fqdn.py#L110) on the `ssl-termination` relation
2. the `fqdn1::proxy` departed hook fired when `fqdn1` was removed causing `proxy` to [set the .update flag](https://github.com/tengu-team/interface-ssl-termination/blob/84d9a9fa846c32dcc921577c5ce9dc88062d5a7a/provides.py#L40)
3. `proxy` is now in the departed hook and enters [get_certificate_request](https://github.com/tengu-team/layer-ssl-termination-proxy/blob/6603287ef0a0d8035da43fd37a594ef3adc961e5/reactive/ssl_termination_proxy.py#L53), eventually getting an empty `cert_requests` because `fqdn1` set `{}` in step 1; the fact that the certs are empty means `ssl-termination.waiting` is never set
4. `proxy` calls [clean_nginx](https://github.com/tengu-team/layer-ssl-termination-proxy/blob/6603287ef0a0d8035da43fd37a594ef3adc961e5/reactive/ssl_termination_proxy.py#L66) which removes all config, but no longer has sufficient flags to subsequently enter `configure_nginx` for remaining valid sites

I think this could be fixed by unconditionally calling `set_flag('ssl-termination.waiting')` in `get_certificate_requests()`, and just let `configure_nginx()` figure out if there are new sites that need to be configured.

Edit: Forgot to mention my workaround:

```bash
juju run --unit proxy/0 -- charms.reactive set_flag ssl-termination.waiting
```

This forced `proxy` back through `configure_nginx()` which setup the missing `g2` and `g3` sites.

**Comment by xannz** [comment]
Thanks for testing. The revisions `ssl-termination-fqdn-5` and  `ssl-termination-proxy-15` are quite old and don't use the recent rewrite. If you want you can try `ssl-termination-proxy-17` and `ssl-termination-fqdn-6`, they are both on the edge channel for now.

**Comment by xannz** [comment]
After testing some more, there is definitely still an issue in de edge channel.

**Comment by xannz** [comment]
This should be fixed in the latest edge version. Could you try again with:
`juju deploy ssl-termination-proxy --channel edge`
`juju deploy ssl-termination-fqdn --channel edge`

**Comment by kwmonroe** [comment]
I'm afraid not; I don't have access to the `edge` channel for either charm.

**Comment by xannz** [comment]
Is there a way I can make them available without pushing them to stable?

**Comment by kwmonroe** [comment]
Yup, there are a few options.  You can view current charm permissions like this:

```bash
charm show ~tengu-team/ssl-termination-[fqdn|proxy] --channel [edge|beta|candidate|stable] perm
```

And then grant permissions to various charm channels. For example, let everyone read the `edge` channel with:

```bash
charm grant ~tengu-team/ssl-termination-fqdn --channel edge --acl read everyone
```

Some teams like to keep the `edge` channel restricted so they have a way to deploy charms under active development without risking an incomplete charm being used by `everyone`. If you'd like to adopt this approach:
- keep the current `edge` channel as-is
- grant `--acl read everyone` to the `beta` and/or `candidate` channels
- release your current edge charms to `beta` and/or `candidate`

Of course, you can also leave everything as-is and grant *me* access to the edge channel like this:

```bash
charm grant ~tengu-team/ssl-termination-fqdn --channel edge --acl read kwmonroe
```

See `grant` and `revoke` sections for more details here: https://discourse.jujucharms.com/t/charm-tools/1180

**Comment by xannz** [comment]
I've granted everyone access to the edge channel.

**Comment by kwmonroe** [comment]
@xannz I upgraded my env to these edge charms:
```
~tengu-team/ssl-termination-fqdn  7
~tengu-team/ssl-termination-proxy 18
```

And all looks good -- expected nginx proxy config remains (or is at least regenerated) when fqdn subordinates depart.  Thanks for the fixes!!

Side note, I noticed something strange with proxy-18:

https://api.jujucharms.com/charmstore/v5/~tengu-team/ssl-termination-proxy-18/archive/config.yaml

Where are those `[string|boolean|int]-option` config opts coming from?  It's not in the source from this repo, so it must be in one of the included layers, but I couldn't find it.  It doesn't hurt anything, but I thought you might want to figure it out before promoting revision 18.
